core/tools.js

```diff
@@ -1182,6 +1182,22 @@
 			return obj;
 		},
 
+		/**
+		 * Finds all spans styled with the given property in the editor contents.
+		 *
+		 * @since 4.15.0
+		 * @param {String} property CSS property which will be used in query.
+		 * @returns {Array} Returns an array of {@link CKEDITOR.dom.element}s.
+		 */
+		getStyledSpans: function( property, source ) {
+			var spans = source.find( 'span[style*=' + property + ']' ).toArray();
+
+			// This is to filter out spans e.g. with background color when we want text color.
+			return CKEDITOR.tools.array.filter( spans, function( span ) {
+				return ( span.getStyle( property ) && span.getText() );
+			} );
```

Comment:
I'm a bit unsure if this function will give expected results as a general function. What if it's expected that the span is empty? Maybe it would be better to just find all spans with `style` attribute and filter the correct property?